### PR TITLE
Typos and branch

### PR DESCRIPTION
**Pull request by hbiede**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.

- Adding non-interactive to branch:delete
- Fixing a few typos

# How

How did you build this feature or fix this bug and why?

Mimicking the format from `channel:delete`

# Test Plan

Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.

Reproducible by running `eas branch:delete --non-interactive "test"` and expecting no prompt

### CODE REVIEW

**Comment by hbiede** [comment]
/changelog new-feature Add `--non-interactive` flag to `branch:delete`

File: packages/eas-cli/src/commands/branch/create.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export default class BranchCreate extends EasCommand {
         message: 'Provide a branch name:',
         initial:
           (await getVcsClient().getBranchNameAsync()) ||
-          `branch-${Math.random().toString(36).substr(2, 4)}`,
+          `branch-${Math.random().toString(36).substring(2, 4)}`,
```
Comment (hbiede) [inline packages/eas-cli/src/commands/branch/create.ts:97]: Deprecated function

File: packages/eas-json/src/types.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export interface EasJson {
     version?: string;
     requireCommit?: boolean;
     appVersionSource?: AppVersionSource;
-    promptToConfigurePushNotfications?: boolean;
+    promptToConfigurePushNotifications?: boolean;
```
Comment (hbiede) [inline packages/eas-json/src/types.ts:24]: If this typo is required to stay to maintain compatibility with another package/library, let me know and I can pull this change

**Comment by wkozyra95** [comment]
Hi thanks for the contribution, but we already have an open PR that adds non-interactive flag to branch:delete https://github.com/expo/eas-cli/pull/1323/files

As for `promptToConfigurePushNotfication` this is already published, so we will need to discuss internally how to handle that, and probably add some fallback to avoid breaking users config.

**Comment by hbiede** [comment]
Handled in #1332

**Pull request closed without merging**